### PR TITLE
Add support for static images

### PR DESCRIPTION
**Pull request by wkozyra95**
Supports:
- PNG
- JPEG
- GIF(only as static image)
- SVG(subset of SVG supported by resvg)

### CODE REVIEW

File: compositor_render/src/transformations/image_renderer.rs [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+    fn download_file(url: &str) -> Result<bytes::Bytes, ImageError> {
+        // TODO: support local files
+        let response = reqwest::blocking::get(url)?;
+        let response = response.error_for_status()?;
+        Ok(response.bytes()?)
+    }
+}
+
+pub struct ImageNode {
```
Comment (wkozyra95) [inline compositor_render/src/transformations/image_renderer.rs:56]: This struct is not needed, but we will want to support images that are not just a static bitmap e.g.
- GIF with animations
- SVG with resolution specified at the update scene level, and not during the transformation registration)

Comment (wkozyra95) [reply compositor_render/src/transformations/image_renderer.rs:56]: outdated

File: compositor_render/src/transformations/image_renderer.rs [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+    image: Arc<Image>,
+    was_rendered: Mutex<bool>,
+}
```
[the remark this answers is not in the document]
Comment (wkozyra95) [reply compositor_render/src/transformations/image_renderer.rs:59]: I did that initially, this was the error I mentioned.
If I moved it there then was_rendered would be a one value per transformation, which would not work if more than one node used it. Also, I wouldn't be refreshed when scene is updated.

[the remark this answers is not in the document]
Comment (WojciechBarczynski) [reply compositor_render/src/transformations/image_renderer.rs:59]: Ahh, right 😃

**Review by WojciechBarczynski — APPROVED** [review]

File: compositor_render/src/transformations/image_renderer.rs [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        // TODO: support local files
+        let response = reqwest::blocking::get(url)?;
+        let response = response.error_for_status()?;
+        Ok(response.bytes()?)
+    }
+}
+
+pub struct ImageNode {
+    image: Arc<Image>,
+    was_rendered: Mutex<bool>,
```
Comment (noituri) [inline compositor_render/src/transformations/image_renderer.rs:58]: Maybe consider using `AtomicBool` instead?

**Pull request merged**